azure-application-insights-spring-boot-starter/src/main/java/com/microsoft/applicationinsights/boot/initializer/SpringBootTelemetryInitializer.java

```diff
@@ -23,9 +23,8 @@
 
   @Override
   public void initialize(Telemetry telemetry) {
-
-    DeviceContext device = telemetry.getContext().getDevice();
-    device.setRoleName(appName);
+    CloudContext cloud = telemetry.getContext().getCloud();
+    cloud.setRole(appName);
```

Comment:
AppMap uses this to create a new Role (element) in UX. Essentially this represents a Microservice in it self. I will come back to why I made this comment here, in another comment in this PR.